x-pack/plugins/transform/server/routes/api/transforms.ts

```diff
@@ -261,45 +264,86 @@ export function registerTransformsRoutes(routeDependencies: RouteDependencies) {
       {
         version: '1',
         validate: {
           request: {
             params: transformIdParamSchema,
             body: putTransformsRequestSchema,
+            query: putTransformsQuerySchema,
           },
         },
       },
-      license.guardApiRoute<TransformIdParamSchema, undefined, PutTransformsRequestSchema>(
-        async (ctx, req, res) => {
-          const { transformId } = req.params;
+      license.guardApiRoute<
+        TransformIdParamSchema,
+        PutTransformsQuerySchema,
+        PutTransformsRequestSchema
+      >(async (ctx, req, res) => {
+        const { transformId } = req.params;
+        const { createDataView, timeFieldName } = req.query;
+
+        const response: PutTransformsResponseSchema = {
+          dataViewsCreated: [],
+          dataViewsErrors: [],
+          transformsCreated: [],
+          errors: [],
+        };
+
+        const esClient = (await ctx.core).elasticsearch.client;
+        await esClient.asCurrentUser.transform
+          .putTransform({
+            // @ts-expect-error @elastic/elasticsearch group_by is expected to be optional in TransformPivot
+            body: req.body,
+            transform_id: transformId,
+          })
+          .then(() => {
+            response.transformsCreated.push({ transform: transformId });
+          })
+          .catch((e) =>
+            response.errors.push({
+              id: transformId,
+              error: wrapEsError(e),
+            })
+          );
 
-          const response: PutTransformsResponseSchema = {
-            transformsCreated: [],
-            errors: [],
-          };
+        if (createDataView) {
+          const { savedObjects, elasticsearch } = coreStart;
+          const dataViewsService = await dataViews.dataViewsServiceFactory(
+            savedObjects.getScopedClient(req),
+            elasticsearch.client.asScoped(req).asCurrentUser,
+            req
+          );
 
-          const esClient = (await ctx.core).elasticsearch.client;
-          await esClient.asCurrentUser.transform
-            .putTransform({
-              // @ts-expect-error @elastic/elasticsearch group_by is expected to be optional in TransformPivot
-              body: req.body,
-              transform_id: transformId,
-            })
-            .then(() => {
-              response.transformsCreated.push({ transform: transformId });
-            })
-            .catch((e) =>
-              response.errors.push({
-                id: transformId,
-                error: wrapEsError(e),
-              })
+          const dataViewName = req.body.dest.index;
+          const runtimeMappings = req.body.source.runtime_mappings as Record<string, RuntimeField>;
+
+          try {
+            const dataViewsResp = await dataViewsService.createAndSave(
+              {
+                title: dataViewName,
+                timeFieldName,
+                ...(isPopulatedObject(runtimeMappings) && isLatestTransform(req.body)
```

Comment:
I know this is carried over from old code, but why do we only apply runtime mappings if it's latest? Why not also for pivot transform?

Reply:
Note with the latest merge of `main` including the recent refactor of the routes into individual files this code moved here: https://github.com/elastic/kibana/pull/160513/files#diff-abc15c191ee8a6304d3346698ac58b5dbbf5d5b0248bf4964151f3556440e314R84-R87